### PR TITLE
Web support

### PR DESCRIPTION
**Pull request by Benny-**
See #14

### CODE REVIEW

File: lib/src/nyxx/Shard.dart [hunk 1]
```diff
@@ -133,12 +131,12 @@ class Shard {
...
             "token": _client._token,
             "properties": <String, dynamic>{
-              "\$os": Platform.operatingSystem,
+              "\$os": operatingSystem,
               "\$browser": "nyxx",
               "\$device": "nyxx",
             },
             "large_threshold": _client._options.largeThreshold,
-            "compress": true
+            "compress": browser ? false : true
```
Comment (l7ssha) [inline lib/src/nyxx/Shard.dart]: you can just do `!browser`

File: lib/src/nyxx/internal/Http.dart [hunk 2]
```diff
@@ -285,8 +285,11 @@ class Http {
   Logger _logger = Logger.detached("Http");
 
   Http._new() {
-    this._headers= {'User-Agent':
-        'DiscordBot (https://github.com/l7ssha/nyxx, ${_Constants.version})'};
+    if(!browser)
```
Comment (l7ssha) [inline lib/src/nyxx/internal/Http.dart]: "Clients using the HTTP API must provide a valid User Agent". [See](https://discordapp.com/developers/docs/reference#user-agent)

File: lib/src/nyxx/internal/Http.dart [hunk 3]
```diff
@@ -296,12 +299,15 @@ class Http {
       bool beforeReady = false,
       Map<String, String> headers = const {},
       String reason}) async {
+    final Map<String, String> _headers = Map.from(this._headers)..addAll(headers);
+    if(!browser)
```
Comment (l7ssha) [inline lib/src/nyxx/internal/Http.dart]: I think web client should be also able to set audit logs

File: lib/src/utils/Binder.dart [hunk 4]
```diff
@@ -2,28 +2,6 @@ import 'package:nyxx/nyxx.dart';
...
-      for(var incl in classRefl.declarations.values.whereType<VariableMirror>()) {
-        if(meta.streamName == incl.simpleName.toString()) {
-          instanceThis.getField(incl.simpleName).reflectee.listen((evnt) {
-            Future.microtask(() => lib.invoke(decl.simpleName, [evnt]));
-          });
-        }
-      }
-    }
-  }
-}
```
Comment (l7ssha) [inline lib/src/utils/Binder.dart]: Why did you remove this completely instead of moving to other place

Comment (Benny-) [reply lib/src/utils/Binder.dart]: It was not referenced. Would you still like it somewhere? If so, where?

File: lib/src/nyxx/internal/Http.dart (hunk 3 above)
Comment (Benny-) [reply lib/src/nyxx/internal/Http.dart]: Setting that header was not allowed due to CORS.

File: lib/src/nyxx/internal/Http.dart (hunk 2 above)
Comment (Benny-) [reply lib/src/nyxx/internal/Http.dart]: Yikes! That is impossible for a web script. I will ask about this in discord api chat.
